### ISSUE TITLE
Support for OS-specific "bundle values"

### ISSUE THREAD

**Issue by application-developer-DA** [burtonageo/cargo-bundle#33]
Bundles for some operating systems may have more options available than for others.

For instance if we're talking about OS X, it has several properties which you can specify in Info.plist such as `CFBundleURLName` (quite a common thing if you want your app to be able to catch the corresponding URI requests), at the moment there is no way to add additional properties to Info.plist as it has a fixed, hardcoded format with only certain (the most common) values.

It would be nice to be able to specify OS-specific bundle properties.

**Comment by mdsteele** [comment]
Some additional OS-specific fields we could consider adding:
* `deb_depends` (list of strings) - this would also take care of issue #25
* `deb_postinst` (string)
* `deb_prerm` (string)
* `ios_interface_orientation` (string)
* `ios_launch_image_file` (string)
* `ios_prerendered_icon` (boolean)
* `ios_requires_persistent_wifi` (boolean)
* `ios_status_bar_hidden` (boolean)
* `ios_status_bar_style` (string)
* `osx_frameworks` (list of strings)
* `osx_high_resolution_capable` (boolean)
* `osx_resources_file_mapped` (boolean)
* `rpm_depends` (list of strings)
* `rpm_postinst` (string)
* `rpm_prerm` (string)

**Comment by virtualritz** [comment]
For macOS, how about adding code that automatically turns `this_name` in `Cargo.toml`'s `
[package.metadata.bundle]` section into `CFBundleThisName` in `Info.plist`?

For example, I need:
`CFBundleSignature`
`CFBundlePackageType`
`CFBundleExecutable`

Asking for each of these and waiting for the devs to update cargo-bundle is not practical.

**Comment by aspenluxxxy** [comment]
Something like `[package.metadata.bundle.deb]` and `[package.metadata.bundle.mac]` would be useful - for example, packaging a systemd .service file in a deb, and a launchd .plist file in an .app. or an .icns file on macOS, .png on Linux, etc etc

**Comment by balanceglove2** [comment]
Seems like this project is pretty much dead, sadly.

**Comment by mdsteele** [comment]
Yeah, I haven't been able to provide any active development on this project lately.  However, PRs are always welcome, and I do still try to keep up with reviewing those.  If there's a feature you need, please go ahead and add it!
